python_modules/libraries/dagster-dbt/dagster_dbt/dbt_project.py

```diff
@@ -48,16 +49,18 @@ def __init__(
         if not self.project_dir.exists():
             raise DagsterDbtProjectNotFoundError(f"project_dir {project_dir} does not exist.")
         self._target_folder = Path(target_folder)
-        self._prepare_command = prepare_command
-        # this is ok if it doesn't exist, will get created by `prepare`
+        self._manifest_cmd = generate_artifacts_command
+        # this is ok if it doesn't exist, will get created during sync
         self._packaged_project_dir = Path(packaged_project_dir) if packaged_project_dir else None
 
         if self._should_use_built_artifacts():
             self._manifest_path = self._current_project_dir.joinpath(
                 self._target_folder, "manifest.json"
             )
```

Comment:
[1]

Reply:
Maybe let it take an "expected_artifacts_path"? 

By default, we'll look in this directory for `manifest.json`.

If the `manifest.json` doesn't exist, we'll blow it away by calling `generate_dbt_artifacts()`. Finally, we can assert that `expected_artifacts_path` is populated afterwards.

Reply:
the other problem is handling the init time implicit generation for local development [2] 

i think it might be workable to shuffle stuff around and add those arguments, but in the end it doesn't seem clearly better to me. What attributes of that approach make it appealing to you?

Subclassing and overriding `generate_artitfacts` is one escape hatch for deeper customization if the common case args we take in the constructor are not sufficient. That or bailing from this class and doing yourself.

Reply:
In general, I want any dbt CLI command to only be specified using `DbtCliResource.cli(...)`.

I don't like the indirection that's happening by specifying `generate_artifacts_command`. At that point, you might as well give the user complete control over how the artifacts are created (i.e. by subclassing). And if they can subclass, then they can just call `DbtCliResource`.

Reply:
Does getting rid of the `generate_artifacts_command` argument and encouraging subclassing for custimization in the doc block of `generate_artifacts` sound good to you?

Reply:
yes